protocol/x/subaccounts/types/expected_keepers.go

```diff
@@ -63,6 +63,13 @@ type PerpetualsKeeper interface {
 		err error,
 	)
 	GetAllPerpetuals(ctx sdk.Context) []perptypes.Perpetual
+	IsPerpetualTradable(
```

Comment:
Currently this is added in `PerpetualsKeeper` interface. An alternative, more generic approach is to add to the parent `ProductKeeper` interface (with method name like `IsUpdatable`.